src/blocks/notmuch.rs

```diff
@@ -0,0 +1,282 @@
+#![allow(non_camel_case_types)]
+
+use chan::Sender;
+use std::time::Duration;
+
+use block::{Block, ConfigBlock};
+use config::Config;
+use de::deserialize_duration;
+use errors::*;
+use scheduler::Task;
+use widget::I3BarWidget;
+use widgets::text::TextWidget;
+use input::{I3BarEvent, MouseButton};
+
+extern crate libc;
+use self::libc::c_char;
+use std::env;
+use std::ffi::CString;
+use std::ptr;
+use std::result;
+use widget::State;
+
+use uuid::Uuid;
+
+pub enum notmuch_query_t{}
+
+pub enum notmuch_database_t{}
+
+// Status codes used for the return values of most functions.
+///
+/// A zero value (SUCCESS) indicates that the function completed without error. Any other value
+/// indicates an error.
+#[repr(C)]
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+pub enum notmuch_status_t {
+    /// No error occurred.
+    SUCCESS = 0,
+    /// Out of memory.
+    OUT_OF_MEMORY,
+    /// An attempt was made to write to a database opened in read-only
+    /// mode.
+    READ_ONLY_DATABASE,
+    /// A Xapian exception occurred.
+    ///
+    /// @todo We don't really want to expose this lame XAPIAN_EXCEPTION
+    /// value. Instead we should map to things like DATABASE_LOCKED or
+    /// whatever.
+    XAPIAN_EXCEPTION,
+    /// An error occurred trying to read or write to a file (this could
+    /// be file not found, permission denied, etc.)
+    FILE_ERROR,
+    /// A file was presented that doesn't appear to be an email
+    /// message.
+    FILE_NOT_EMAIL,
+    /// A file contains a message ID that is identical to a message
+    /// already in the database.
+    DUPLICATE_MESSAGE_ID,
+    /// The user erroneously passed a NULL pointer to a notmuch
+    /// function.
+    NULL_POINTER,
+    /// A tag value is too long (exceeds TAG_MAX).
+    TAG_TOO_LONG,
+    /// The `notmuch_message_thaw` function has been called more times
+    /// than `notmuch_message_freeze`.
+    UNBALANCED_FREEZE_THAW,
+    /// `notmuch_database_end_atomic` has been called more times than
+    /// `notmuch_database_begin_atomic`.
+    UNBALANCED_ATOMIC,
+    /// The operation is not supported.
+    UNSUPPORTED_OPERATION,
+    /// The operation requires a database upgrade.
+    UPGRADE_REQUIRED,
+    /// There is a problem with the proposed path, e.g. a relative path
+    /// passed to a function expecting an absolute path.
+    PATH_ERROR,
+    /// One of the arguments violates the preconditions for the
+    /// function, in a way not covered by a more specific argument.
+    NOTMUCH_STATUS_ILLEGAL_ARGUMENT,
+}
+
+#[repr(C)]
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+pub enum notmuch_database_mode_t {
+    NOTMUCH_DATABASE_MODE_READ_ONLY = 0,
+    NOTMUCH_DATABASE_MODE_READ_WRITE,
+}
+
+#[link(name = "notmuch")]
```

Comment:
What will happen if this is not linked correctly?

Reply:
If the package is unavailable, then the project fails to build, that's what led me to ask about a plugin system.